src/commcare_cloud/commands/ansible/ansible_playbook.py

```diff
@@ -106,6 +108,20 @@ def get_limit():
             return ()
 
     def ansible_playbook(environment, playbook, *cmd_args):
+        required_ansible_version = "4.2.0"
+        venv_version = subprocess.run(["pip", "show", "ansible"], stdout=subprocess.PIPE, universal_newlines=True)
+        ansible_version = str(venv_version.stdout).split("\n")[1].split(" ")[1]
+        try:
+            if os.environ['VIRTUAL_ENV']:
+                if (version.parse(ansible_version) < version.parse(str(required_ansible_version))):
+                    puts(color_error(f"The version of ansible you have installed ({ansible_version}) is no longer supported."))
+                    puts(color_notice(f"To upgrade from ansible {ansible_version} to {required_ansible_version} or above you will first have to uninstall the current version (due to an idiosyncratic issue)"))
+                    puts(color_code("  pip uninstall ansible"))
+                    puts(color_notice("before re-installing the supported version using your standard method."))
+                    return 2
+        except KeyError:
+            puts(color_notice("please activate the cchq virtualenv."))
```

Comment:
Few issues here:

1. The subprocess call is unnecessary, use `ansible.__version__` instead.
2. The `try`/`except` block is unnecessary and will do the wrong thing if `os.environ['VIRTUAL_ENV']` evaluates to false.
3. This function should return immediately if the virtualenv is not active.

```suggestion
        if 'VIRTUAL_ENV' not in os.environ:
            puts(color_notice("please activate the cchq virtualenv."))
            return 2
        min_ansible_version = "4.2.0"
        if version.parse(ansible.__version__) < version.parse(min_ansible_version):
            puts(color_error(f"The version of ansible you have installed ({ansible.__version__}) is no longer supported."))
            puts(color_notice(f"To upgrade from ansible {ansible.__version__} to {min_ansible_version} or above you will first have to uninstall the current version (due to an idiosyncratic issue)"))
            puts(color_code("  pip uninstall ansible"))
            puts(color_notice("before re-installing the supported version using your standard method."))
            return 2
```

Reply:
I think the point of the most recent commits is that `ansible.__version__` is returning a different version number from the one we're looking for. Prem says more about this in the ticket, but I think the deal is that it's returning the "ansible-core" version number instead of the version number associated with the `ansible` pip package, which is what we're looking for. (He was using `ansible.__version__` here previously and faced this issue.)

The other points stand though

Reply:
I looked at the ticket and was not able to understand the problem with `ansible.__version__`, so for now I remain unconvinced (more details requested).

Reply:
I see that since 4.0 (ansible-core 2.10.0), `ansible.__version__` reports the core version, and the version used to install from pip is completely different. I haven't figured out the reason for that yet, and it's super weird to `pip install` one version and then get something completely different from the package's `__version__`! Indeed, the `4.x` version does not appear to be anywhere in the ansible source code in `site-packages`. Why do you do that, ansible?

However, the [ansible 2.10 porting guide](https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.10.html#porting-guide-for-v2-10-0) says the uninstall is necessary when upgrading from versions prior to 2.10, so I think we can use `2.10.0` as the minimum version string (which can be compared to `ansible.__version__`), before which we will show the uninstall message. I have updated the suggestion above.

Reply:
Thanks @millerdev this is a great find